app/util/job.ts

```diff
@@ -194,11 +191,36 @@ export async function resumeAndSaveJob(
         await workflowStep.save(tx);
       }
     }
-    job.resume();
+    if (jobFnName === 'resume') {
+      job.resume();
+    } else if (jobFnName === 'skipPreview') {
+      job.skipPreview();
+    }
     await job.save(tx);
   });
 }
 
+/**
+ * Resume a paused job then save it.
+ *
+ * @param jobID - the id of job (requestId in the db)
+ * @param _logger - the logger to use for logging errors/info
+ * @param username - the name of the user requesting the resume - null if the admin
+ * @param token - the access token for the user
+ * @throws {@link ConflictError} if the job is already in a terminal state.
+ * @throws {@link NotFoundError} if the job does not exist or the job does not
+ * belong to the user.
+ */
+export async function resumeAndSaveJob(
+  jobID: string,
+  _logger: Logger,
+  username?: string,
+  token?: string,
+
+): Promise<void> {
+  await updateTokenAndChangeState(jobID, username, token, 'resume');
```

Comment:
Like I said earlier, I think it would be better if this function just called `updatedToken` without the change state part using a shared transaction and then did the 'resume' part and job saving here.

Reply:
That's where I started, but the transaction does not start at the beginning of the code so it makes sense that it does the entire thing.

I really want to just pass in the function to call but from a bunch of searching I cannot find a way you can do that with a method on an object (thought that maybe Symbol construction could work). Anyway what I ended up with is the best I could come up with. I'll give it more thought today.

Reply:
Oh - if the common function returns a database transaction I could use the transaction to make the status change in the calling function... Feels a little clunky to return a database connection but it seems like passing in a string to tell it what to do is also clunky. Not sure which would be better.

Reply:
Figured out how to do my original intent by passing in a function that takes a job as an argument. It's no longer clunky.